torch/ao/quantization/pt2e/quant_lift_up_pass.py

```diff
@@ -0,0 +1,93 @@
+import torch
+from torch.fx.node import map_arg
+from torch.fx.passes.infra.pass_base import PassBase, PassResult
+
+
+# Only for per tensor quant since permute may changes the channel idx
+_QUANTIZE_OPS = [
+    torch.ops.quantized_decomposed.quantize_per_tensor.default,
+    torch.ops.quantized_decomposed.quantize_per_tensor.tensor,
+]
+
+_VIEW_OPS = [
+    torch.ops.aten.transpose.int,
+    torch.ops.aten.permute.default,
+    torch.ops.aten.view.default,
+]
+
+
+class QuantLiftUp(PassBase):
```

Comment:
we discussed this before, so this is one way to do it, the other way is just to annotate view, permute, transpose with SharedQuantizationSpec between output and input, and we'll fuse the (dq -> view -> q) pattern in lowering to be a single view op that works with quantized tensor. And last time I think you preferred this approach. I also feel observing these ops will be better since you'll be able to simulate quantization the same way the other ops are doing, also good for QAT as well. Is there a reason to do this lifting instead of the observation approach?

Reply:
Thanks for the comment. Yean, I have another discussion with @chunyuan-w, she mentions a case for the downside of this `observation approach`. For example a pattern as
```
linear
view
fp32_op doesn't support int8
```
1. The lift up pass approach will generate pattern as 
```
dq
linear
view
fp32_op
```
2. The observation approach will generate pattern as
```
dq
linear
q
dq
view
fp32_op
```
In this case, the observation approach will do redundant q/dq after the linear op.

Also cc @jgong5 for this discussion.

Reply:
Ah yes, seems like forcing the quantization for the ops like view, transpose, permute would introduce side-effect on accuracy and performance like the case @leslie-fang-intel pointed out. @jerryzh168 any thoughts on this?

Reply:
I believe these two passes are equivalent in terms of how much quantization patterns or ways of quantization they can express, the only difference is 2 is more explicit than 1.

in this specific case, what you can do is to decide whether to observe view or not based on the annotation of previous operator, if the previous operator (linear) is observed, then we'll observe it as well, otherwise we'll just skip annotation. this is more explicit than the lifting pass I think.

Reply:
> in this specific case, what you can do is to decide whether to observe view or not based on the annotation of previous operator, if the previous operator (linear) is observed, then we'll observe it as well, otherwise we'll just skip annotation. 

Thanks and I think that's exactly the same recipe of `view` we mentioned above in https://github.com/pytorch/pytorch/pull/122777#discussion_r1543944541 with the `observation approach`. In which, it brings a extra pair of `Quant/Dequant` at the output of `linear` comparing to `lift up pass` for this specific case.

To avoid it in `observation approach`, we may have to check the following up node of `view` whether it can be quantized (beside input node is observed or not). Then we will decide to observer `view` or not. Any thoughts?

Reply:
I see, so to summarize, whether the output of linear should be observed or not depends on whether view is followed by a quantizable op?

Reply:
this makes sense I think, in that case it's probably better to do in lowering to not complicate the logic in annotation. but I'm wondering if you could do this in intel lowering code base, instead of in quantization code base?

Reply:
> I see, so to summarize, whether the output of linear should be observed or not depends on whether view is followed by a quantizable op?

Thanks for the comment. In current recipe, if linear followed by a quantizable op, we will annotate the input of this quantizable operator (which is also the output of this linear). And for view, the logic in annotation will be complicate.